src/yetibot/commands/github.clj

```diff
@@ -137,36 +137,32 @@
                   (f/unparse date-formatter datetime))))
       (str n " is not a number"))))
 
+(defn format-release
+  "Displays information about a release"
+  [release org-name repo]
+  (if (nil? (:status release))
+    (let [tag (:tag_name release)
+          author (get-in release [:author :login])
+          published-at (-> "YYYY-MM-dd'T'HH:mm:ssZ"
+                           (f/formatter)
+                           (f/parse (:published_at release)))]
+      (format "Release version, tagged: %s from %s/%s, was published on %s by %s" tag org-name repo
+              (f/unparse date-hour-formatter published-at) author))
+    (format "No release version info found for %s/%s" org-name repo)))
+
 (defn show-latest-release-info-cmd
   "gh releases show <org>/<repo-name> # retrieve info about the latest release on a Github repository"
   {:yb/cat #{:util :info}}
   [{[_ org-name repo] :match}]
-  (let [latest-release (gh/latest-releases org-name repo)
-        status (:status latest-release)]
-    (if (nil? status)
-      (let [tag (:tag_name latest-release)
-            author (get-in latest-release [:author :login])
-            published-at (:published_at (-> "YYYY-MM-dd'T'HH:mm:ssZ"
-                                            (f/formatter)
-                                            (f/parse (:published_at latest-release))))]
-        (format "%s/%s latest version, tagged: %s, was published on %s by %s" org-name repo tag
-                (f/unparse date-hour-formatter published-at) author))
-      (format "No release version info found for %s/%s" org-name repo))))
+  (if-let [release (gh/latest-releases org-name repo)]
+    (format-release release org-name repo)))
```

Comment:
Looks good but do we need to handle the case where `release` is nil and tell the user that we couldn't find the release? I see you already have that error handling in `format-release`, but it'll never get called of the release is nil right?

Reply:
Actually, `gh/latest-releases` and `gh/release-by-tag` always return a _json result_ when called, even though there are no releases associated with a repository. So `release` always have a value.
In `format-release`, I then check for the presence of the `:status` field in the json response, to check whether information about the release has been found or not.

```
(gh/latest-releases "kaffein" "clojure-applied")
=>
{:request-time 1378,
 ...
 ...
 :status 404,
 :length -1,
 ...,
 :trace-redirects []}
```

```
(gh/release-by-tag "kaffein" "clojure-applied" "1.0.0-RC1")
=>
{:request-time 997,
 ...
 :status 404,
 ...,
 :trace-redirects []}
```

_Note_ : I probably may have used `let` here instead of `if-let`, since I can always expect to have a result from the call.

On the other hand though, the call to `gh/releases` may return `nil`. I should have used `if-let` here instead.

```
(gh/releases "kaffein" "clojure-applied")
=> nil
```

Reply:
Yep I'd switch that to `let` if you always expect it to be non-nil.